### ISSUE TITLE
Notification Ideas?

### ISSUE THREAD

**Issue by Justinclark87** [jeanpaulsio/whiztutor-mobile#37]
Drop one signal?  Any ideas on replacements?

**Comment by Johnsalzarulo** [comment]
Long term: I am leaning toward dropping in-app notifications but NOT until we support SMS responses to messages. That's the ideal solution in my ming. 

Short Term: I think we should go with [Expo's push notification system.](https://docs.expo.io/versions/v18.0.0/guides/push-notifications.html), this takes a LOT of configuration from Apple's app store (Via developer account) and the play store (android). Honestly, just managing that whole security chain. 

I'll have to update our backend to support Expo and JP will have to integrate on the front end. 

I'm not sure who can handle the certificate generation bull. Probably me or JP. I know that Dale and John R had tons of issues on this.

**Comment by jeanpaulsio** [comment]
I think it would be ideal to lean on SMS messages not unlike apps like Postmates 

But users definitely need to be able to respond to those messages

**Comment by Johnsalzarulo** [comment]
I agree I think sms that handles replies is BEST. But - unless we are willing to pay for the "forwarding" numbers and set that up then SMS doesn't make a lot of sense. 😕

UNLESS - @jeanpaulsio - Are we able to "Deep Link" into your app? How hard is that? Because we could send a shortlink with every notification that opens up that part of the app... messages, bookings, etc.  That could be a good middle ground between "Push" notifications and "SMS" until we can **do** SMS the **right** way.

**Comment by Johnsalzarulo** [comment]
I reeeeeealy think [this new service from Twilio](https://www.twilio.com/proxy) will solve this cheaper. This solves exactly what we are trying to do - I have applied for the "early access" program but not with a @whiztutor email address. I don't have or want access to the one that was set up for me. It might help if someone applied to the "early access" program with a real @whiztutor address.

**Comment by jeanpaulsio** [comment]
HMMMMMMMMMMMM very interesting idea with the deep linking. 

I had to do some deep linking on WUPHF when i was having individual accounts authenticate to twitter and then come back into the app

however i dont know if i could route to a specific part of the app. it might just reopen the app

**Comment by jeanpaulsio** [comment]
more complex deep linking will require me to detach

**Comment by Johnsalzarulo** [comment]
We don't even need "complex" deep linking. In my mind we would be ok with just 3 Index screens: Messages, Bookings and Account. I'm not sure any notification would take any other screen.

**Comment by Johnsalzarulo** [comment]
I could add it to this end point we're talking about doing, on the user
auth return.

We could have a field called like redirect to

That way when a user opens the app if there's a pending notification they
are redirected to that page

Or we could only have the Deep linking open the notifications page, then
the notification can link to the page.

**Comment by jeanpaulsio** [comment]
yeah i was thinking something like:

*when the app loads it checks for an authentication token in local storage
* IF authentication token exists, send a GET request to the new notification endpoint
* if theres anything in this notification queue, send them to the notification page 


the only thing i'm not 100% sure about is if this will work if the app is in the background and not completely closed

**Comment by jeanpaulsio** [comment]
the deep linking that i can do with Expo (as far as i know) will literally just open the app - not open the app to any 1 specific page

so if they are on the "my bookings" page and get an SMS with a deep link, click that deep link, it will just open the app back up and they will land on "my bookings"

**Comment by jeanpaulsio** [comment]
so i'm not 100% sure deep linking will refresh the app's state if the app is already running in the background. i'll have to look into this later
